src/UICrafter.Client/Components/AppViewButtons.razor

```diff
@@ -34,26 +65,90 @@
     [Parameter]
     public required EventCallback<MudDropContainer<UIComponent>?> DropContainerChanged { get; set; }
 
-    private void SaveAppView()
+    AppView test = new AppView();
+
+    private async Task SaveAppView()
     {
-        throw new NotImplementedException();
+        Visible = true;
+
+        var serializedContent = UIComponentList.ToByteString();
+
+        AppView.Content = serializedContent;
+
+        try
+        {
+            if (AppView.Id == 0)
+            {
+                await AppViewRepository.CreateAppViewAsync(AppView);
+                Console.WriteLine("Called Create");
+            }
+            else
+            {
+                await AppViewRepository.UpdateAppViewAsync(AppView);
+                Console.WriteLine("Called Update");
+            }
+        }
+        catch (RpcException ex)
+        {
+            Console.WriteLine($"Error while saving AppView: {ex.Status.Detail}");
+        }
+
+        Visible = false;
     }
 
-    private void DeleteAppView()
+    private async Task DeleteAppView()
     {
-        throw new NotImplementedException();
+        Visible = true;
+
+        try
+        {
+            if (AppView.Id != 0)
+            {
+                await AppViewRepository.DeleteAppViewByIdAsync(AppView.Id);
+
+                ClearAppView();
+
+                AppView = new AppView();
+            }
+        }
+        catch (RpcException ex)
```

Comment:
You should not be throwing RpcExceptions. Use something else, maybe argument exception or invalid operations